src/lib.rs

```diff
@@ -1,20 +1,20 @@
 #[macro_export]
 macro_rules! pyo3_built {
-    ($py: ident, $info: ident) => {{
+    ($py: ident, $info: ident, "intro") => {
```

Comment:
I think it is a bit confusion to put "into" and "finish" in the same place as the build information keys. Looking at how this is used, is it actually necessary to generate this code via recursively invoking the macro? Wouldn't it be much simpler to just inline the uses and the return value as the invocation sites in the "custom build" pattern?

Reply:
Yes, in general the implementation-specific code is hidden in [a rule that uses `__impl`](https://stackoverflow.com/questions/50647701/whats-the-significance-of-impl-in-a-macro-pattern). It would be an error to let the user call the macro with `pyo3_built!("intro", "host")` even though the macro would accept it just fine.